src/util/converter/factory.js

```diff
@@ -0,0 +1,146 @@
+ * @returns {converter} - Returns the text converter, ready for use.
+ * @export 'util/converter/factory'
+ */
+export default function converterFactory(builtinProcessors = [], builtinConfig = {}) {
+    let processors = [];
+
+    // find the index of the named processor
+    const findProcessor = name => processors.findIndex(processor => processor.name === name);
+
+    /**
+     * @typedef {object} converter
+     */
+    const converter = {
+        /**
+         * Converts a text with respect to the registered converter processors.
+         * @param {string} text - The text to convert.
+         * @param {object} [config] - An optional config object that may contain processor specific configuration.
+         *                            It will be merged with the possible builtin config.
+         * @returns {string} - Returns the converted text.
+         */
+        convert(text, config = {}) {
+            const localConfig = Object.assign({}, builtinConfig, config);
+            for (const processor of processors) {
+                text = processor.convert.call(converter, text, localConfig);
+            }
+
+            return text;
+        },
+
+        /**
+         * Registers a converter processor.
+         * @param {converterProcessor} processor - The converter processor to register.
+         * @returns {converter} - Chains the instance.
+         */
+        register(processor) {
+            if (!validateProcessor(processor)) {
+                throw new TypeError('The given processor is not valid!');
+            }
+
+            if (findProcessor(processor.name) > -1) {
+                throw new TypeError(`The processor "${name}" is already registered!`);
+            }
+
+            processors.push(processor);
+
+            return this;
+        },
+
+        /**
+         * Unregisters a converter processor.
+         * @param {string|converterProcessor} name - The name of the processor to remove.
+         * @returns {converter} - Chains the instance.
+         */
+        unregister(name) {
+            if ('object' === typeof name) {
+                name = name.name;
+            }
+
+            processors = processors.filter(processor => processor.name !== name);
+
+            return this;
+        },
+
+        /**
+         * Removes all converter processors.
+         * @returns {converter} - Chains the instance.
+         */
+        clear() {
+            processors = [];
+
+            return this;
+        },
+
+        /**
+         * Tells whether a converter processor is registered or not.
+         * @param {string} name - The name of the processor to check.
+         * @returns {boolean} - Returns `true` if the converter processor is registered ; returns `false` otherwise.
+         */
+        isRegistered(name) {
+            return findProcessor(name) > -1;
+        }
+    };
+
+    for (const processor of builtinProcessors) {
```

Comment:
Yes, it could be if the code was used in several places, to keep it DRY. As this is the only place, no need to overcomplicate with unnecessary nesting :)